chainer/functions/pooling/pooling_2d.py

```diff
@@ -48,6 +47,8 @@ def check_type_forward(self, in_types):
         )
 
     def forward_gpu(self, x):
+        self._used_cudnn = True
```

Comment:
Pooling2D is a base class and `forward_gpu` is overridden by each implementation. This function is only called when the implementation chooses cuDNN.